subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/ScriptCacheKey.java

```diff
@@ -37,15 +39,17 @@ public boolean equals(Object o) {
 
         ScriptCacheKey key = (ScriptCacheKey) o;
 
-        return classLoader.equals(key.classLoader)
+        return classLoader.get() != null
```

Comment:
⭕️ It doesn't really matter, but for symmetry this should return true if both classloaders are null.